app/assets/javascripts/discourse/tests/integration/components/user-menu/notification-item-test.js

```diff
@@ -0,0 +1,200 @@
+import { module, test } from "qunit";
+import { setupRenderingTest } from "discourse/tests/helpers/component-test";
+import { exists, query, queryAll } from "discourse/tests/helpers/qunit-helpers";
+import { render, settled } from "@ember/test-helpers";
+import { deepMerge } from "discourse-common/lib/object";
+import { NOTIFICATION_TYPES } from "discourse/tests/fixtures/concerns/notification-types";
+import Notification from "discourse/models/notification";
+import { hbs } from "ember-cli-htmlbars";
+import I18n from "I18n";
+
+function getNotification(overrides = {}) {
+  return Notification.create(
+    deepMerge(
+      {
+        id: 11,
+        user_id: 1,
+        notification_type: NOTIFICATION_TYPES.mentioned,
+        read: false,
+        high_priority: false,
+        created_at: "2022-07-01T06:00:32.173Z",
+        post_number: 113,
+        topic_id: 449,
+        fancy_title: "This is fancy title &lt;a&gt;!",
+        slug: "this-is-fancy-title",
+        data: {
+          topic_title: "this is title before it becomes fancy <a>!",
+          display_username: "osama",
+          original_post_id: 1,
+          original_post_type: 1,
+          original_username: "velesin",
+        },
+      },
+      overrides
+    )
+  );
+}
+
+module(
+  "Integration | Component | user-menu | notification-item",
+  function (hooks) {
+    setupRenderingTest(hooks);
+
+    const template = hbs`<UserMenu::NotificationItem @item={{this.notification}}/>`;
+
+    test("pushes `read` to the classList if the notification is read", async function (assert) {
+      this.set("notification", getNotification());
+      this.notification.read = false;
```

Comment:
I realise that having a codebase mixing glimmer/not glimmer forces us to think how the object is created to know if we need set or not.